Core/Src/throttle.c

```diff
@@ -35,58 +35,53 @@ void throttle_read(void *argument) {
-        static float apps2_fix_mul[APPS_MATRIX_LENGTH] = {APPS2_FIX_MUL};
-        static float apps2_fix_add[APPS_MATRIX_LENGTH] = {APPS2_FIX_ADD};
+        apps_ref apps1_ref = {APPS1_REF, APPS1_FIX_MUL, APPS1_FIX_ADD};     //valores de referencia e parametros
+        apps_ref apps2_ref = {APPS2_REF, APPS2_FIX_MUL, APPS2_FIX_ADD};     //para o calculo da porcentagem.
 
-        apps2_throttle_percent = throttle_calc(APPS2,apps2_value,apps2_fix_mul,apps2_fix_add,APPS2_MAX_PERCENT);    //calcula a porcentagem do pedal a partir do APPS2
-        apps1_calc = throttle_calc(apps2_throttle_percent,apps1_value,apps1_fix_mul,apps1_fix_add,APPS1_MAX);       //calcula a porcentagem esperada do APPS1 a partir da porcentagem calculada do APPS2
+        apps1_throttle_percent = throttle_calc(apps1_value,apps1_ref);              //calcula a porcentagem do pedal a partir do APPS1
+        apps2_throttle_percent = throttle_calc(apps2_value,apps2_ref);              //calcula a porcentagem do pedal a partir do APPS2
+        throttle_percent = (apps1_throttle_percent + apps2_throttle_percent)/2;     //a porcentagem final do pedal é a media dos dois apps
 
-        set_global_var_value(BRAKE_STATUS, (BSE > BRAKE_ACTIVE));
-        set_global_var_value(THROTTLE_STATUS, (apps2_throttle_percent > 0));
+        set_global_var_value(BRAKE_STATUS, (bse > BRAKE_ACTIVE));
+        set_global_var_value(THROTTLE_STATUS, (throttle_percent > 0));
 
         log_data(ID_BRAKE, get_global_var_value(BRAKE_STATUS));
 
+        check_for_errors(is_there_APPS_error, APPS_ERROR_FLAG);                                                          //verifica a plausabilidade dos APPSs
+        check_for_errors_with_timeout(is_there_APPS_error, APPS_ERROR_FLAG, tim_APPS_errorHandle, APPS_ERROR_TIMER);     //verifica a plausabilidade do BSE
+        check_for_errors_with_timeout(is_there_SU_F_error, SU_F_ERROR_FLAG, tim_SU_F_errorHandle, SU_F_ERROR_TIMER);     //verifica se a placa de freio está em curto
 
-        //verifica a plausabilidade do BSE
-        check_for_errors(is_there_BSE_error, BSE_ERROR_FLAG);
-        //verifica a plausabilidade dos APPSs
-        check_for_errors_with_timeout(is_there_APPS_error, APPS_ERROR_FLAG, tim_APPS_errorHandle, APPS_ERROR_TIMER);
-        //verifica se a placa de freio está em curto
-        check_for_errors_with_timeout(is_there_SU_F_error, SU_F_ERROR_FLAG, tim_SU_F_errorHandle, SU_F_ERROR_TIMER);
-
-        uint16_t message = apps2_throttle_percent;
+        uint16_t message = throttle_percent;
         osMessageQueuePut(q_throttle_controlHandle, &message, 0, 0U);
 
         osDelay(THROTTLE_DELAY);
     }
 }
 
-uint16_t throttle_calc(uint16_t APPS, uint16_t *value, float *fix_mul, float *fix_add, uint16_t MAX) {
-    if(APPS < 0)
+uint16_t throttle_calc(uint16_t apps_value, apps_ref ref) {
```

Comment:
passa esse `apps_ref` por referência
```suggestion
uint16_t throttle_calc(uint16_t apps_value, const apps_ref* ref) {
```
se não vc vai copiar aquela struct grandona toda vez q chama essa função, mas não precisa. Isso vai ser bem ruim de performance